aten/src/ATen/native/TensorConversions.cpp

```diff
@@ -1673,24 +1664,14 @@ Tensor sparse_compressed_to_sparse_bsc(const Tensor& self, IntArrayRef blocksize
                 "[blocksize=", blocksize,"] is not implemented.");
     return sparse_compressed_clone(self, blocksize, "to_sparse_bsc");
   }
-  AT_ERROR(
-      "sparse_compressed_to_sparse_bsc expected SparseBsr or SparseBsc layout but got ",
-      self.layout());
-  return self;
-}
-
-Tensor sparse_compressed_to_sparse_csc(const Tensor& self, c10::optional<int64_t> dense_dim_opt) {
-  if (dense_dim_opt.has_value()) {
-    AT_ERROR("sparse_compressed_to_sparse_csc conversion does not support specifying number of dense dimensions");
-  }
-  if (self.layout() == kSparseCsr) {
-    return sparse_compressed_to_flipped(self, c10::nullopt, "to_sparse_csc");
-  }
   if (self.layout() == kSparseCsc) {
-    return sparse_compressed_clone(self, c10::nullopt, "to_sparse_csc");
+    TORCH_CHECK(self.dim() == 2 + self.dense_dim(),
+                "to_sparse_bsc: conversion from Csc to Bsc for batched inputs is not implemented.");
+
+    return _compressed_to_block_compressed_cpu<kSparseBsc>(self.cpu(), blocksize).to(self.device());
```

Comment:
We generally do not want to perform device transfers silently like this. We would either add the required kernel so the conversion can be performed on GPU, or warn here.

Reply:
@amjames I think the problem here is that we have only CPU implementation of the kernel (copied from scipy, IIRC) and the aim of this PR is to generalize CSR->BSR for the CSC->BSC case. Implementing CUDA kernel would be great indeed but I think it would be a good follow-up PR candidate.

Reply:
Indeed - as mentioned above, this PR is just extending CSR->BSR conversion, as is, to cover CSC->BSC too.  But I've just added a `TORCH_WARN` in case when the code executing on a non-CPU device.